scene/gui/line_edit.cpp

```diff
@@ -255,11 +255,28 @@ void LineEdit::_gui_input(Ref<InputEvent> p_event) {
 				selection.creating = true;
 
 			} else {
-				if (b->is_doubleclick() && selecting_enabled) {
-					selection.enabled = true;
-					selection.begin = 0;
-					selection.end = text.length();
-					selection.doubleclick = true;
+				if (selecting_enabled) {
+					if (!b->is_doubleclick() && (OS::get_singleton()->get_ticks_msec() - selection.last_dblclk) < 600) {
+						// Triple-click select all.
+						selection.enabled = true;
+						selection.begin = 0;
+						selection.end = text.length();
+						selection.doubleclick = true;
+						selection.last_dblclk = 0;
+					} else if (b->is_doubleclick()) {
+						// Double-click select word.
+						Vector<Vector2i> words = TS->shaped_text_get_word_breaks(text_rid);
+						for (int i = 0; i < words.size(); i++) {
+							if (words[i].x < cursor_pos && words[i].y > cursor_pos) {
+								selection.enabled = true;
+								selection.begin = words[i].x;
+								selection.end = words[i].y;
+								selection.doubleclick = true;
+								selection.last_dblclk = OS::get_singleton()->get_ticks_msec();
```

Comment:
```suggestion
								selection.last_dblclk = OS::get_singleton()->get_ticks_msec();
								cursor_pos = selection.end;
```

I would move the cursor to the end of the word, for consistency with TextEdit and other text editors.